### DISCUSSION TITLE
Anyone else got lost on the configuration step?

### DISCUSSION THREAD

**Discussion by jaamrek** [bellingcat/auto-archiver#70] [Q&A] [answered]
I'm trying to set up on a mac and not really familiar with python etc. I'm got everything else installed and the json service file. Not really sure what the next step is here. 

What do I need to do with the config.yaml file? 

Any help is appreciated.

**Comment by djhmateer** [comment]
Dear Jaamrek

I've been working with the archiver for a good while now and contribute to the project (I work with https://osr4rightstools.org/auto-archiver)

So reading between the lines, I'd say this is a tall order for you to get it going as it takes some time. If you've got patience, then I'm happy to help! 

If you'd like me to set you up with your own demo sheet on our commercial offering, I'm happy to do that too.

Regards
Dave
dave@hmsoftware.co.uk

**Comment by msramalho** [comment]
Great that you're able to help Dave. 
From your description, @jaamrek it seems you already:
* installed everything needed (python and its libraries)
* have a config.yaml with your own configurations
The next step should simply be to call the auto-archiver from the command line and let it know which configuration file to use with the `--config config.yaml` setting. To do that open a terminal and make sure your working directory is where the auto-archiver files are (you can test by entering `pwd` into the console to check the folder path and then call `python auto_archive.py --config config.yaml`

**Comment by jaamrek** [comment]
Hi both, thanks for you replies. 

I was working using the instruction video where the call for windows appears to be a bit different. I'm on mac. When I run `python auto_archive.py --config config.yaml`I get this message: ModuleNotFoundError: No module named 'ffmpeg'. 

I've downloaded FFmpeg. I've tried having it in different folders, re-downloaded, tried different download methods (brew and from the website). 

Any ideas?

Thanks so much.

**Comment by msramalho** [comment]
Can you try these and report back:

Since the error seems to be form python itself and you're [on  mac](https://stackoverflow.com/questions/48653837/installing-ffmpeg-on-mac):
1. type `python --version` to check that `python` refers to python3, if you get python 2.X, try `python3 --version`
2. if you used pipenv check with `pipenv run python --version` to see if it's using 3.X
3. if you used `pip` call it with `python -m pip...` instead of directly `pip` (if in 1. you saw that `python3` was the correct one, use it here)
4. now with either pip or pipenv do `pipenv install -r requirements.txt` or `python -m pip install -r requirements`
5. try again
6. if that does not work, try using pip or pipenv to manually install the ffmpeg library: `pipenv install ffmpeg-python`

to test it along the way you can type `python` or `python3` (whichever is 3.x on your computer) and it will open an interpreter, in the interpreter type `import ffmpeg` + enter and check if you still get the error, it should disappear once the installation is correct.

---

On a side note, we plan to release a dockerized version soon which would mean it would be 10x easier to run the archiver and you'll not face these issues, but it can take a few weeks still.

**Comment by msramalho** [comment, marked as answer]
The new documentation should really help with understanding the configuration process, please refer to it here: https://auto-archiver.readthedocs.io/en/latest/index.html


On top of that, we built a visual tool to edit your orchestration.yaml files, use it here: https://auto-archiver.readthedocs.io/en/latest/installation/config_editor.html


NB: the current video should get a revamp as so much has changed in the code, likewise the article will also be updated in the future to reflect that and other changes in the AA ecosystem such as the release of https://github.com/bellingcat/auto-archiver-api
